js/ui/grid_core/ui.grid_core.validating.js

```diff
@@ -362,6 +364,10 @@ const ValidatingController = modules.Controller.inherit((function() {
                 if(showEditorAlways && editingController.isCellOrBatchEditMode() && editingController.allowUpdating({ row: parameters.row })) {
                     editIndex = editingController._addEditData({ key: parameters.key, oldData: parameters.data });
                 }
+
+                if(this.option('editing.editRowKey') === parameters.key) {
```

Comment:
```suggestion

                if(editindex < 0 && this.option('editing.editRowKey') === parameters.key) {
```

**_let editData;_** should be removed from **createValidator**. It is better to put **editData** declarations directly to **getValue** and **dataGetter** callbacks.

```js
const getValue = () => {
    const editData = editingController.getEditDataByKey(validationData?.key);
    ...
}
...
dataGetter: function() {
    const editData = editingController.getEditDataByKey(validationData?.key);
     ...
},
```

Reply:
>  if(editindex < 0 && this.option('editing.editRowKey') === parameters.key) {

We don't need this because `editIndex < 0` condition is met on the 357 line.

Reply:
> let editData; should be removed from createValidator.

Ok.

Reply:
>We don't need this because editIndex < 0 condition is met on the 357 line.

You are right, but it may confusing if the following code will set editIndex to a value that is greater or equal to 0:
```js
if(showEditorAlways && editingController.isCellOrBatchEditMode() && editingController.allowUpdating({ row: parameters.row })) {
    editIndex = editingController._addEditData({ key: parameters.key, oldData: parameters.data });
}
```
and then, reset editIndex to 0 under the following condition:

```js
if(this.option('editing.editRowKey') === parameters.key) {
    editIndex = 0;
}
```

Reply:
Ok, I will move index reset before this condition.